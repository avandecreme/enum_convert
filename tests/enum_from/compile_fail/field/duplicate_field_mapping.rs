use enum_convert::EnumFrom;

enum Source {
    Struct { x: i32 },
}

#[derive(EnumFrom)]
#[enum_from(Source)]
enum Target {
    #[enum_from]
    Struct {
        // Should be #[enum_from(Source::Struct.x)]
        #[enum_from(Source::Struct.x, Source::Struct.y)]
        a: i32,
    },
}

fn main() {}
