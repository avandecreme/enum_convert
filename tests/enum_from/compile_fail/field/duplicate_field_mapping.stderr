error: Multiple mapping found for source enum `Source`
  --> tests/enum_from/compile_fail/field/duplicate_field_mapping.rs:13:9
   |
13 |         #[enum_from(Source::Struct.x, Source::Struct.y)]
   |         ^
